common/src/main/spark-3.3/org/apache/comet/shims/ShimBatchReader.scala

```diff
@@ -19,8 +19,12 @@
 
 package org.apache.comet.shims
 
+import scala.collection.mutable
+
+import org.apache.spark.executor.TaskMetrics
 import org.apache.spark.sql.catalyst.InternalRow
 import org.apache.spark.sql.execution.datasources.PartitionedFile
+import org.apache.spark.util.AccumulatorV2
 
 object ShimBatchReader {
```

Comment:
nit: the shiimmed method is for getting an accumulator from TaskMetrics and does not seem specific to `BatchReader`, so maybe it should be in a specific `ShimTaskMetric` object?

Reply:
I believe we tend to put the methods in the class they are used. Anyway, let me move this to `BatchReader` and avoid the code duplication.